cli/tests/unit/test_marathon_pod.py

```diff
@@ -56,25 +68,58 @@ def resource_reader(path):
 
 
 def _assert_pod_remove_invoked_successfully(pod_id, force):
-    def resource_reader(path):
-        assert False, "should not be called"
     marathon_client = create_autospec(marathon.Client)
 
-    subcmd = main.MarathonSubcommand(resource_reader, lambda: marathon_client)
+    subcmd = main.MarathonSubcommand(
+        resource_reader=_unused_resource_reader,
+        create_marathon_client=lambda: marathon_client)
     returncode = subcmd.pod_remove(pod_id, force)
 
     assert returncode == 0
     marathon_client.remove_pod.assert_called_with(pod_id, force)
 
 
 def _assert_pod_remove_propagates_exceptions_from_remove_pod(exception):
-    def resource_reader(path):
-        assert False, "should not be called"
     marathon_client = create_autospec(marathon.Client)
     marathon_client.remove_pod.side_effect = exception
 
-    subcmd = main.MarathonSubcommand(resource_reader, lambda: marathon_client)
+    subcmd = main.MarathonSubcommand(
+        resource_reader=_unused_resource_reader,
+        create_marathon_client=lambda: marathon_client)
     with pytest.raises(exception.__class__) as exception_info:
         subcmd.pod_remove('does/not/matter', force=False)
 
     assert exception_info.value == exception
+
+
+@patch('dcoscli.marathon.main.emitter', autospec=True)
+def _assert_pod_show_invoked_successfully(emitter, pod_json):
+    marathon_client = create_autospec(marathon.Client)
+    marathon_client.show_pod.return_value = pod_json
+
+    subcmd = main.MarathonSubcommand(
+        resource_reader=_unused_resource_reader,
+        create_marathon_client=lambda: marathon_client)
+
+    returncode = subcmd.pod_show(pod_json['id'])
+
+    assert returncode == 0
+    marathon_client.show_pod.assert_called_with(pod_json['id'])
+    emitter.publish.assert_called_with(pod_json)
+
+
+def _assert_pod_show_propagates_exceptions_from_show_pod(exception):
+    marathon_client = create_autospec(marathon.Client)
+    marathon_client.show_pod.side_effect = exception
+
+    subcmd = main.MarathonSubcommand(
+        resource_reader=_unused_resource_reader,
+        create_marathon_client=lambda: marathon_client)
+    with pytest.raises(exception.__class__) as exception_info:
```

Comment:
Yes, but it depends on what your idea of a "correct marathon error message" is. [These tests](https://github.com/dcos/dcos-cli/blob/60ce2d47b899a4654faf801feb5b29fe3b0e1d3c/tests/test_marathon.py#L217-L359) cover the error messages for the statuses mentioned in the Marathon `/v2/pods` API docs: 400, 401, 403, 404, 409, and 422. Except for 400 and 409, the CLI doesn't do anything special with the error JSON, it just extracts some fields and prints them on the console. If we want nicer messages for specific statuses, let's figure out what they should be and we can write tests for them.